src/renderer/components/cluster-manager/clusters-menu.tsx

```diff
@@ -149,16 +148,15 @@ export class ClustersMenu extends React.Component<Props> {
           )}
         </div>
         <div className="extensions">
-          {globalPageMenuRegistry.getItems().map(({ id: menuItemId, title, url, components: { Icon } }) => {
-            const registeredPage = globalPageRegistry.getById(menuItemId);
+          {globalPageMenuRegistry.getItems().map(({ title, url, components: { Icon } }) => {
+            const registeredPage = globalPageRegistry.getByUrl(url);
```

Comment:
This also won't work with extra defined `:placeholder`-s in some `routePath` for registered page.

Reply:
Example:
```ts
export default class ExampleExtension extends LensRendererExtension {
  clusterPages: Interface.PageRegistration[] = [
    {
      id: "example-extension-id",
      routePath: "/items/:id",
      components: {
        Page: () => <ExamplePage extension={this}/>,
      }
    }
  ]

  clusterPageMenus: Interface.PageMenuRegistration[] = [
    {
      id: this.clusterPages[0].id,
      title: "Hello World",
      url: "/items/1",
      components: {
        Icon: ExampleIcon
      }
    }
  ]
}
```